cmd/juju/controller/listcontrollers.go

```diff
@@ -83,8 +83,7 @@ func (c *listControllersCommand) Run(ctx *cmd.Context) error {
 		return errors.Annotate(err, "failed to list controllers")
 	}
 	if len(controllers) == 0 && c.out.Name() == "tabular" {
-		ctx.Infof("%s", modelcmd.ErrNoControllersDefined)
-		return nil
+		return errors.Errorf("%s", modelcmd.ErrNoControllersDefined)
```

Comment:
Just use 
```go
return errors.Trace(modelcmd.ErrNoControllersDefined)
```

The errors package gives the ability to trace or annotate existing errors.